src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -793,7 +793,7 @@ public void run() {
                                     result = oauthRequest.getToken(endingUrl);
                                     Logger.v(TAG, "OnActivityResult processed the result. "
                                             + authenticationRequest.getLogInfo());
-                                } catch (Exception exc) {
+                                } catch (IOException exc) {
```

Comment:
Same here. Should it also catch the IllegalArgumentException and AuthenticationException throwed by getToken()? @weijjia